### ISSUE TITLE
Unhandled rejection error when trying to swallow dropdb() failure

### ISSUE THREAD

**Issue by RavenHursT** [olalonde/pgtools#17]
I was trying to implement what this issue was asking for at a higher level: https://github.com/olalonde/pgtools/issues/3 

I wrote this out:

```javascript
import pgtools from 'pgtools'
import CONFIG from './db-config.json'

function handleDropErrors (err) {
  console.dir(err.message.indexOf(`does not exist`), {colors: true, depth: 3})
  if (err.message.indexOf(`does not exist`) === -1) {
    console.error(err)
  } else {
    console.error(`got here`, err)
    // createDb()
    return Promise.resolve()
  }
}

function createDb() {
  // calls pgtools.createdb()
}

pgtools.dropdb(CONFIG, CONFIG.database)
  .then(createDb, handleDropErrors)
  .catch(handleDropErrors)
```

But I keep getting the following stack trace:
```
Unhandled rejection error: database "mydb" does not exist
    at Connection.parseE (/Users/mmarcus/dev/emma-exercise/node_modules/pgtools/node_modules/pg/lib/connection.js:567:11)
    at Connection.parseMessage (/Users/mmarcus/dev/emma-exercise/node_modules/pgtools/node_modules/pg/lib/connection.js:391:17)
    at Socket.<anonymous> (/Users/mmarcus/dev/emma-exercise/node_modules/pgtools/node_modules/pg/lib/connection.js:129:22)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    at TCP.onread (net.js:548:20)
```

AFACT, I _am_ handling the error, I'm just choosing to swallow it if the error is a specific error.

What's even more odd, is that w/ both native _and_ Bluebird promises.. this kind of logic seems to work just fine for "swallowing" a promise rejection.

Native: https://repl.it/JgPz
Bluebird: https://repl.it/JgQa/1

I'm at a loss at this point :-(

**Comment by RavenHursT** [comment]
Ok... figured out what the problem here is...

`pgtools`, if you pass in a `database` property on the config, tries to connect to said DB: 
https://github.com/olalonde/pgtools/blob/master/index.js#L45-L48

I have a `db-config.json` that I had the name of my DB that I was trying to drop/create, in.

So this allows me to get around that:

```javascript
import pgtools from 'pgtools'
import CONFIG from './db-config.json'

// Have to remove database from config or else `pgtools` throws error
const noDbConfig = Object.assign({}, CONFIG, {
  database: undefined
})

pgtools.dropdb(noDbConfig, CONFIG.database).then(createDb)
  .catch(handleDropErrors)

function handleDropErrors (err) {
  if (err.message.indexOf(`does not exist`) === -1) {
    throw(err)
  } else {
    createDb()
  }
}

function createDb() {
  pgtools.createdb(noDbConfig, CONFIG.database)
    .then(console.log)
    .catch(err => {
      throw(err)
    })
}
```

Wondering if it makes any sense for `pgtools` to ever try to connect to a database in the config, when most likely, you're going to be creating/dropping said DB anyway?

**Comment by bcoe** [comment]
@RavenHursT we're using Bluebird promises in pgtools, so definitely curious about why errors aren't behaving the way you'd expect.

If I understand properly, `db-config.json` has `mydb` populated? which then failed because you're attempting to use `pgtools` to create `mydb`?

**Comment by RavenHursT** [comment]
Yeah... basically.. AFAICT, `pdgtools` is just a "wrapper" to connect to some "default" db in postgres, so that you can create and drop DBs since you can't just "connect" to postgres w/o connecting to an existing DB. Right?

So what happened is, because my config contained the name of the DB I wanted to drop/create, the underlying `pg` library was trying to connect to a DB that didn't exist yet (because it was the DB that I was trying to create in the first place), rather than defaulting to the `postgres` DB as these lines here indicate: https://github.com/olalonde/pgtools/blob/master/index.js#L45-L48

Basically, was just confusing because I was first trying to `dropdb()`, but was being told that the DB doesn't exist.. not because of the `DROP` query being run (it never was), but because of the initial attempt to connect to the nonexistent DB.

Wondering if there's a way to make the "default to connecting to this DB" a little less confusing than just assuming the default connection DB will be `postgres || config.database`, since most likely, `config.database` will usually come from some system-level config and will contain the database name of the DB that a user will be wanting to drop/create?

**Comment by adriaandotcom** [comment]
I want to start with a clean slate. So closing this issue as it's from 2017. Please reopen any issue if it's still a problem at the moment.
